src/Halogen/Aff/Driver.purs

```diff
@@ -1,109 +1,47 @@
 module Halogen.Aff.Driver
-  ( RenderSpec
-  , runUI
+  ( runUI
+  , hydrateUI
+  , module Halogen.Aff.Driver.RenderImplementation
```

Comment:
Can we keep the `runUI` and `hydrateUI` implementations directly in this module rather than indirectly via an "implementation" module?

In general, if code needs to be moved to another module (to avoid import cycles, or because it's internal and users should not import it directly) I would prefer that it is moved to an `Internal` namespace like `Halogen.Aff.Driver.Internal`.

In this case it doesn't seem clear to me that the code needs to be moved to another module. I'd rather just keep both implementations in this module unless you feel strongly that that isn't a good idea.

That has the added advantage of keeping the diffs a little more readable for reviewers -- if this module is expanded then we won't have to read through `runUI` code that wasn't changed, just moved, which will help us get your work reviewed sooner.

Reply:
I'll check, the code between renderImpl and hydrateImpl will be shared

Reply:
I have moved and split code to the Implementation directory